lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -1241,22 +1241,48 @@ func (c *ContainerProxy) Remove(vc *viccontainer.VicContainer, config *types.Con
 // once the said container has been removed. It fetches a list of volumes that are joined
 // to at least one other container, and calls the portlayer to remove this container's
 // anonymous volumes if they are dangling. Errors, if any, are only logged.
-func removeAnonContainerVols(pl *client.PortLayer, cID string, volumes map[string]struct{}) {
+func removeAnonContainerVols(pl *client.PortLayer, cID string, vc *viccontainer.VicContainer) {
+	// NOTE: these strings come in the form of <volume id>:<destination>:<volume options>
+	volumes := vc.Config.Volumes
+	// NOTE: these strings come in the form of <volume id>:<destination path>
+	namedVolumes := vc.HostConfig.Binds
+
+	// assemble a mask of volume paths before processing binds. MUST be paths, as we want to move to honoring the proper metadata in the "volumes" section in the future.
+	namedMaskList := make(map[string]struct{}, 0)
+	for _, entry := range namedVolumes {
+		fields := strings.SplitN(entry, ":", 2)
+		if len(fields) >= 2 {
+			log.Errorf("Invalid entry in the HostConfig.Binds metadata section for container %s: %s", cID, entry)
+		}
+		destPath := fields[1]
```

Comment:
this can still cause a panic if `len(fields) == 1`. SplitN doesn't guarantee you that number of fields, it's a maximum.
https://play.golang.org/p/_Pw8WCCmtaA